app/models/concerns/orchestration.rb

```diff
@@ -119,6 +119,13 @@ def skip_orchestration_for_testing?
     Rails.env.test?
   end
 
+  def with_orchestration_toggle(disabled = true, &block)
+    skip_orchestration! if disabled && SETTINGS[:unattended]
+    yield
+  ensure
+    enable_orchestration! if disabled && SETTINGS[:unattended]
+  end
```

Comment:
I'd probably name it like this:

    without_orchestration_if(!SETTINGS[:enable_orchestration_on_fact_import]) do
    end